### ISSUE TITLE
Cannot see variable name input dialog when using plugin for enhanced UI

### ISSUE THREAD

**Issue by rish987** [andrewferrier/debugprint.nvim#80]
I use the plugin [noice.nvim](https://github.com/folke/noice.nvim) which enhances the UI for functions like `vim.fn.input`. I have found that, when using the recommended keymaps with `expr = true`, the input dialog doesn't show up -- I'm still able to input, and after I press enter the debug string shows up, but while typing I don't get any visual feedback. I was able to fix this by removing `expr = true` and calling `vim.fn.feedkeys`, so my mappings look like this:
```lua
vim.keymap.set("n", "<Leader>dv", function()
  vim.fn.feedkeys(require('debugprint').debugprint({ ignore_treesitter = true, variable = true }))
end)
```
I was wondering if there is a particular reason you would recommend *against* doing the mappings like this, but if you think it's okay, then perhaps it's worth mentioning in the README for people who have a similar issue (would be happy to make a PR).

**Comment by andrewferrier** [comment]
@rish987 thanks for these thoughts and your other contributions. The `expr = true` is needed to make [dot-repeat](https://www.vikasraj.dev/blog/vim-dot-repeat) work, essentially. So far I've not found a different approach for this. Maybe you don't use dot repeat which is why you haven't noticed this.

I'm not aware of any specific problems with your approach, although I haven't tried it/used it. If it works for you for now that's fine, but I think the `expr = true` is ugly enough, I don't think I want to confuse the documentation further in this area. Ultimately what I would like to do is refactor `debugprint` so that keymappings can be changed with simple configuration, so that issues like don't arise (although in your case you'd then no longer be able to use your workaround, which is concerning to me). I also want to look into `vim.ui.input()` rather than `vim.fn.input()`, which is the more modern approach (see [issue](https://github.com/andrewferrier/debugprint.nvim/issues/41)), but that's a non-trivial change.

tl;dr: I'm not changing anything right now in this area, I don't see any problem with your workaround for now, but I'm not sure I want to officially support it/document it until there's a cleaner approach. If you have any suggestions, I'm definitely open to them...

**Comment by andrewferrier** [comment]
@rish987 Hi - I think I've fixed this issue now. Since [this refactor](https://github.com/andrewferrier/debugprint.nvim/commit/ee9d6ffa00709f90c51c69afe913fafa9400c9ed), `debugprint.nvim` now configures the keymappings for you using configuration. I've now switched them internally to `feedkeys` rather than `expr` (for most of the keybindings) and it has also fixed this problem with `noice` (I think, I don't use it regularly, but I tried installing it and was able to recreate your issue, and it appears now fixed).

I'd be interested in any further feedback you have. The issue should be fixed on the latest development release (HEAD of `main` branch).

Thanks for bringing this to my attention!

**Comment by rish987** [comment]
Yep, I just updated and can confirm that it's working perfectly (with dot-repeat)! Thank you so much!

**Comment by andrewferrier** [comment]
@rish987 great, that's good to know. Thanks again for the feedback.
